src/client.ts

```diff
@@ -730,77 +727,62 @@ export class SpvWalletClient {
   async SendToRecipients(recipients: Recipients, metadata: Metadata): Promise<Transaction> {
     const draft = await this.DraftToRecipients(recipients, metadata);
-    const finalized = this.SignTransaction(draft);
+    const finalized = await this.SignTransaction(draft);
     return this.RecordTransaction(finalized, draft.id, metadata);
   }
 
   /**
    * Finalize and sign the given draft transaction
    *
-   * @param {DraftTransaction} draftTransaction Draft transaction object
+   * @param {DraftTx} draftTransaction Draft transaction object
    * @return {string} Final transaction hex
-   */
-  SignTransaction(draftTransaction: DraftTransaction): string {
+   */  async SignTransaction(draftTransaction: DraftTx): Promise<string> {
     if (!this?.xPriv) {
       throw new ErrorNoXPrivToSignTransaction();
     }
 
-    const xPriv = this.xPriv as bsv.HDPrivateKey;
-    const txDraft: bsv.Transaction = new bsv.Transaction(draftTransaction.hex);
+    const xPriv = this.xPriv as HD;
+    const txDraft: Transaction = Transaction.fromHex(draftTransaction.hex);
 
-    // sign the inputs
-    const privateKeys: bsv.PrivateKey[] = [];
     draftTransaction.configuration.inputs?.forEach((input, index) => {
+      let hdWallet: HD | undefined;
       if (input.destination) {
         const dst = input.destination;
         // derive private key (m/chain/num)
-        let privKey = xPriv.deriveChild(dst.chain).deriveChild(dst.num);
+        hdWallet = xPriv.deriveChild(dst.chain).deriveChild(dst.num);
 
         if (dst.paymail_external_derivation_num != null) {
           // derive private key (m/chain/num/paymail_num)
-          privKey = privKey.deriveChild(dst.paymail_external_derivation_num);
+          hdWallet = hdWallet.deriveChild(dst.paymail_external_derivation_num);
         }
-
-        privateKeys.push(privKey.privateKey);
       }
 
       // small sanity check for the inputs
       if (
-        input.transaction_id != txDraft.inputs[index].prevTxId.toString('hex') ||
-        input.output_index != txDraft.inputs[index].outputIndex
+        input.transaction_id != txDraft.inputs[index].sourceTXID ||
+        input.output_index != txDraft.inputs[index].sourceOutputIndex
       ) {
         throw new ErrorTxIdsDontMatchToDraft(this.logger, input, index, txDraft.inputs[index]);
       }
 
-      // @todo add support for other types of transaction inputs
-      txDraft.inputs[index] = new bsv.Transaction.Input.PublicKeyHash({
-        prevTxId: input.transaction_id,
-        outputIndex: input.output_index,
-        script: new bsv.Script(input.script_pub_key),
-        output: new bsv.Transaction.Output({
-          script: new bsv.Script(input.script_pub_key),
-          satoshis: input.satoshis,
-        }),
-      });
+      if (hdWallet !== undefined) {
+        txDraft.inputs[index].unlockingScriptTemplate = new P2PKH().unlock(hdWallet.privKey, "single", false, input.satoshis, new P2PKH().lock(input.destination?.address as string));
```

Comment:
I made a suggestion in PR https://github.com/bitcoin-sv/spv-wallet-js-client/pull/936
how to get rid of the type assertion here and make it a little bit clear

Reply:
I applied your PR, thank you!